scheduler/src/cook/mesos/scheduler.clj

```diff
@@ -590,7 +590,9 @@
   (for [{:keys [tasks leases]} matches
         :let [offers (mapv :offer leases)
               slave-id (-> offers first :slave-id :value)]
-        ^TaskAssignmentResult task tasks
+        ^TaskAssignmentResult task (->> tasks
+                                        ;; sort-by makes match transaction deterministic
+                                        (sort-by (comp :db/id :job #(.getRequest %))))
```

Comment:
How much does the `sort` cost computationally? Given that we want to optimize schedule times, should we make this configurable?